### ISSUE TITLE
Switch off the new overreactive interface jump to end of output behaviour?

### ISSUE THREAD

**Issue by sorenwacker** [jupyterlab/jupyterlab#14878]
Since updated jupyter-lab recently the view sometimes suddenly jumps to the bottom of the output of the next cell. I am not sure what exaclty that triggers, but it might be intended as a new 'feature'. I find this behaviour rather disorienting and would prefer to scroll manually. Is there an option to switch these new behaviour off?

I just merged some cells and all of a sudden the view jumps three to five cells below? Maybe it is a bug?

Version 4.0.3

**Comment by krassowski** [comment]
Sounds like a bug. Could you possibly record a screencast with demonstration/provide exact steps to reproduce?

**Comment by sorenwacker** [comment]
Will try.

**Comment by sorenwacker** [comment]
Ok, what happens is that, when a cell is executed, the view jumps to the next cell, which gets aligned at the top of the page. This way I don't see the output of the just executed cell. And I have to scroll up every time to see it.

**Comment by krassowski** [comment]
Do you mean during execution with shift + enter, but ctrl + enter works like you would expect? (in any case this is something that should be fixed).

**Comment by krassowski** [comment]
Reproducer:

JupyterLab 3.x

![jupyterlab-3-x](https://github.com/jupyterlab/jupyterlab/assets/5832902/a4267c2e-ac2f-4bca-9ffe-df84fb88b103)

JupyterLab 4.0.5:

![jupyterlab-4](https://github.com/jupyterlab/jupyterlab/assets/5832902/5892f95d-df2c-4c98-a135-4569ade442e5)

**Comment by krassowski** [comment]
Notes:
1) Upon cell change [`handleState()`](https://github.com/jupyterlab/jupyterlab/blob/897418f0f25ffbbee3aa06a62770a53246823f85/packages/notebook/src/actions.tsx#L2152-L2168) invokes `scrollToItem()` with `smart` strategy; the logic in  `getOffsetForIndexAndAlignment` tries to show the bottom edge of the cell when using `smart` strategy; this is not what users expect. But why it does not show the bottom edge in reproducer?
2) There are 3 calls to `scrollToItem`; the first one comes from `handleState()`, following two are a result of cells changing size as the execution result (in the reproducer the number two, the result of 1+1) is first cleared and then added back:
   ![image](https://github.com/jupyterlab/jupyterlab/assets/5832902/03b7332b-eaba-4be3-a262-cc187585861f)
   the following 2 calls lead to the seemingly random final positioning - somewhere in between the previous and current cell.

**Comment by hversteegzepp** [comment]
This still seems to be happening to me in 4.0.9.
The screen recording below was with a fresh virtual environment with just Jupyterlab, though it did still use my user settings from other installs.
![bug](https://github.com/jupyterlab/jupyterlab/assets/48432477/061b5d72-8cc9-4957-8d20-088e858740ac)

**Comment by krassowski** [comment]
@hversteegzepp can you try changing windowing mode in Notebook settings to "full"?

**Comment by gorkamunoz** [comment]
> @hversteegzepp can you try changing windowing mode in Notebook settings to "full"?

Hi, just wanted to point that some bugs I had in my notebook were solved when changing the window mode from defer to full as pointed out here. 

Aside of the problem opened in this issue, I was also having problems when using the table of content, where clicking in the TOC on the different headings would not send me to that part of the notebook. Instead, the notebook would stay in the same position but the cell of the heading I clicked but open for edit (i.e. as if I would click on the cell itself).

I couldn't find an opened issue related to this, but now it seems to be solved by changing the window mode. Let me know if you want me to open an issue with more details on this, or its some known behaviour.
